src/task.c

```diff
@@ -409,6 +409,7 @@ static void ctx_switch(jl_task_t *lastt)
             // early free of stkbuf back to the pool
             jl_release_task_stack(ptls, lastt);
         }
+        JL_PROBE_RT_FINISH_TASK(lastt);
```

Comment:
As @kpamnany said you probably want https://github.com/JuliaLang/julia/blob/master/src/task.c#L231?